### PR TITLE
Add a new code sample for the exclusive job execution feature

### PR DESCRIPTION
**Pull request by metanet**

### CODE REVIEW

File: job-management/src/main/java/ExclusiveJobExecution.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+        // Both Job handles refer to the same job
+        assert job1.getId() == job2.getId();
+
+        try {
+            // If I try to submit the same job with newJob(), I will get an exception
+            instance1.newJob(p, jobConfig);
+            assert false;
+        } catch (JobAlreadyExistsException ignored) {
+            System.out.println();
```
Comment (viliam-durina) [inline job-management/src/main/java/ExclusiveJobExecution.java]: Should be `System.out.println("Caught exception: " + ignored);

File: job-management/src/main/java/ExclusiveJobExecution.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        try {
+            job1.join();
+        } catch (CancellationException ignored) {
+        }
+
+        // Now, I can submit a new job with the same name because there is no active job with that name
+        Job job3 = instance1.newJobIfAbsent(p, jobConfig);
+
+        // This is a new job instance
+        assert job3.getId() != job3.getId();
```
Comment (viliam-durina) [inline job-management/src/main/java/ExclusiveJobExecution.java]: You compare `job3` to itself

File: job-management/src/main/java/ExclusiveJobExecution.java (hunk 1 above)
Comment (metanet) [reply job-management/src/main/java/ExclusiveJobExecution.java]: fixed

File: job-management/src/main/java/ExclusiveJobExecution.java (hunk 2 above)
Comment (metanet) [reply job-management/src/main/java/ExclusiveJobExecution.java]: this is a copy paste error. fixed it

**Pull request merged**